### PR TITLE
save settings in json instead of xml

### PR DESCRIPTION
**Pull request by akashters**
- use nlohmann_json library to save settings in json format instead of xml format
- remove boost dependency since it's only being used for xml parsing and no longer needed

### CODE REVIEW

File: src/main_screen.cpp [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 #include <imgui-SFML.h>
 #include <imgui.h>
 #include <iostream>
+#include <libremidi/config.hpp>
 #include <libremidi/libremidi.hpp>
+#include <nlohmann/json.hpp>
 #include <string>
 #include <vector>
 
+#define SETTINGS_FILE_NAME "settings.json"
```
Comment (shriramters) [inline src/main_screen.cpp]: Use typed constexpr

```c++
constexpr std::string_view settings_file_name = "settings.json";
```

File: src/preferences.hpp [hunk 2]
```diff
@@ -38,7 +37,16 @@ struct Preferences {
...
+
+class SettingsFileDoesntExistException : public std::exception {
+  private:
+    std::string message;
+
+  public:
+    SettingsFileDoesntExistException() : message("Settings file doesn't exist at the given path") {}
+
+    const char* what() const throw() { return message.c_str(); }
+};
```
Comment (shriramters) [inline src/preferences.hpp]: Personally I would keep it simple and throw `std::runtime_error("file doesn't exist")`

File: src/utils.hpp [hunk 3]
```diff
@@ -4,6 +4,7 @@
 #include "chord_db.hpp"
 #include <algorithm>
 #include <iostream>
+#include <optional>
```
Comment (shriramters) [inline src/utils.hpp:8]: good catch

File: CMakeLists.txt
Comment (shriramters) [inline CMakeLists.txt:1]: Is it necessary to format CMakeLists like this? is this being done by clang format?

My flatpak manifest applies a patch on this which will have to be completely rewritten now if this is formatted

File: src/main_screen.cpp (hunk 1 above)
Comment (akashdoppalapudi) [reply src/main_screen.cpp]: Thanks. I replaced #define with a constexpr

File: src/preferences.hpp (hunk 2 above)
Comment (akashdoppalapudi) [reply src/preferences.hpp]: I used a runtime_error and removed the custom exception, also moved the preferences related code to preferences.cpp thus removing clutter from the main_screen.cpp

File: CMakeLists.txt
Comment (akashdoppalapudi) [reply CMakeLists.txt:1]: I guess it's okay to keep it this way

**Comment by akashdoppalapudi** [comment]
I also think it's better to separate declarations and implementations in utils.hpp with a utils.cpp file.
We should consider it.

**Review by shriramters — APPROVED** [review]
LGTM 
Great work
and also, thanks

**Pull request merged**